pkg/auth/roles_authz.go

```diff
@@ -0,0 +1,104 @@
+package auth
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/environments"
+	shared "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared"
+	arrayUtils "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared/utils/arrays"
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+	"gopkg.in/yaml.v2"
+)
+
+var _ environments.ConfigModule = (*AdminRoleAuthZConfig)(nil)
+
+// RolesConfiguration is the configuration of required roles per HTTP method of the admin API.
+type RolesConfiguration struct {
+	HTTPMethod string   `yaml:"method"`
+	RoleNames  []string `yaml:"roles"`
+}
+
+// RoleConfig represents the role configuration.
+type RoleConfig []RolesConfiguration
+
+// AdminRoleAuthZConfig is the configuration of the role authZ middleware.
+type AdminRoleAuthZConfig struct {
+	Enabled         bool
```

Comment:
This value is not used anywhere outside this file, is the intention to enable / disable roles authz check based on this variable? If so, should this be used in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/4b2405a45f1444cf3b591c1cf75c6f04872e6d44/pkg/auth/roles_authz_middleware.go#L42-L57 as well, and only perform the roles check when this variable is set to `true`. 

If not, I think we can remove this boolean flag altogether.

Reply:
will remove the flag. thanks